index.js

```diff
@@ -68,8 +68,10 @@ function getOptions(uri, o, method) {
     }
 
     // Default to binary requests (return buffer)
-    if (!o.encoding) {
+    if (o.encoding === undefined) {
         o.encoding = null;
+    } else {
+        o.encodingProvided = true;
```

Comment:
Minor: Maybe make this _encodingProvided to emphasize that it's a private thing?